ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -629,6 +636,19 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 		return nil
 	}
 
+	if features.Enabled(features.RayClusterStatusConditions) {
+		if suspendStatus == rayv1.RayClusterSuspended {
+			if instance.Spec.Suspend != nil && !*instance.Spec.Suspend {
```

Comment:
Do we need to return errors (`errResuming` or `errSuspending`) here? Can we return `nil` directly? We can directly infer the cluster is suspending, or suspended, or resuming in `calculateStatus`. Removing the errors can also avoid handling special cases in `calculateStatus`.